cloudbuild.yaml

```diff
@@ -0,0 +1,24 @@
+# See https://cloud.google.com/cloud-build/docs/build-config
+# For more information about Image pushing refer to https://github.com/kubernetes/test-infra/blob/master/config/jobs/image-pushing/README.md
+timeout: 3600s
+
+options:
+  substitution_option: ALLOW_LOOSE
+
+steps:
+  - name: 'gcr.io/cloud-builders/bazel'
+    env:
+    - IMAGE_REGISTRY=gcr.io
+    - IMAGE_REPO=k8s-image-staging
+    - IMAGE_TAG=$_GIT_TAG
+    args:
+      - run
+      - //cmd/cloud-controller-manager:publish
```

Comment:
I think you need to plumb in the GCR credentials from prow if  you use this bazel target. I worked around it previously by just using the bazel target to build an image locally and then using docker to auth and push the image, see https://github.com/kubernetes/cloud-provider-gcp/pull/377/files for reference.

Reply:
@BenTheElder can you help us here?

Reply:
I don't know if they are inherited by the prow job or something

Reply:
cloudbuilds in your GCP staging project should have permission to push.

the prow trusted cluster we permit kicking off a cloudbuild from (in a restricted way) which will then run in that project.

Reply:
My current understanding is that `gcloud auth configure-docker` just works on cloudbuilds, but bazel requires separate config for push access. Not a blocker, we can merge and iterate